### PR TITLE
Add pronouns

### PR DESCRIPTION
**Pull request by manuthecoder**
This PR allows users to add pronouns to their profile from their settings. These pronouns appear in event team pages

### CODE REVIEW

**Comment by sampoder** [comment]
I feel like we don't really need to implement this, we're using it purely for display purposes? I don't know, I want to keep HCB functional.

**Comment by garyhtou** [comment]
I wonder if this would help operations

**Comment by northeastprince** [comment]
It's a common convention to put pronouns in parentheses after a name, and I don't see any reason users can't do that if they want to. [It's a hook; let's save the towel racks for when they're worth the tradeoff.](https://world.hey.com/jason/hooks-towel-bars-and-software-14c66c8c)

File: app/views/users/edit.html.erb [hunk 1]
```diff
@@ -38,6 +38,10 @@
         <p class="h5 muted mt0 mb1">If you go by a different name.</p>
         <%= form.text_field :preferred_name, placeholder: @user.full_name.presence || "FiFi Hacks", maxlength: 30 %>
       </div>
+      <div class="field">
+        <%= form.label :pronouns %>
+        <%= form.select :pronouns, ["Prefer not to say", "She/Her", "He/Him", "They/Them", "Other"] %>
+      </div>
```
Comment (sampoder) [inline app/views/users/edit.html.erb]: This should probably be a string entry not a dropdown

File: app/views/users/edit.html.erb [hunk 2]
```diff
@@ -38,6 +38,10 @@
         <p class="h5 muted mt0 mb1">If you go by a different name.</p>
         <%= form.text_field :preferred_name, placeholder: @user.full_name.presence || "FiFi Hacks", maxlength: 30 %>
       </div>
+      <div class="field">
+        <%= form.label :pronouns %>
+        <%= form.text_field :pronouns, placeholder: "Pronouns" %>
```
Comment (northeastprince) [inline app/views/users/edit.html.erb]: What's the point of the placeholder if there's a label above it?

Comment (manu-gurudath) [reply app/views/users/edit.html.erb]: Just wanted to fill in some awkward space - I changed the text a little bit so it isn't the exact same as the label above

Comment (sampoder) [reply app/views/users/edit.html.erb]: Can you make it `they/them/theirs`?

File: app/models/user.rb [hunk 3]
```diff
@@ -17,6 +17,7 @@
 #  phone_number_verified    :boolean          default(FALSE)
 #  preferred_name           :string
 #  pretend_is_not_admin     :boolean          default(FALSE), not null
+#  pronouns                 :string
```
Comment (garyhtou) [inline app/models/user.rb]: validate length of `pronouns` is less than 20-ish characters

File: app/views/organizer_positions/_organizer_position.html.erb [hunk 4]
```diff
@@ -9,6 +9,7 @@
       </div>
       <div class="flex flex-col justify-center">
         <p class="h2 bold m0"><%= organizer_position.user.name %></p>
+        <p class="h5 bold m0"><%= organizer_position.user.pronouns == "Prefer not to say" ? "" : organizer_position.user.pronouns %></p>
```
Comment (garyhtou) [inline app/views/organizer_positions/_organizer_position.html.erb]: I'm not sure if we need this `organizer_position.user.pronouns == "Prefer not to say"` check.


```suggestion
        <% if organizer_position.user.pronouns.present? %>
          <p class="h5 bold m0"><%= organizer_position.user.pronouns %></p>
        <% end %>
```

File: app/views/users/edit.html.erb [hunk 5]
```diff
@@ -38,6 +38,10 @@
         <p class="h5 muted mt0 mb1">If you go by a different name.</p>
         <%= form.text_field :preferred_name, placeholder: @user.full_name.presence || "FiFi Hacks", maxlength: 30 %>
       </div>
+      <div class="field">
+        <%= form.label :pronouns %>
```
Comment (garyhtou) [inline app/views/users/edit.html.erb]:
```suggestion
        <%= form.label :pronouns, "Pronouns (optional)" %>
```

File: app/views/users/edit.html.erb (hunk 2 above)
Comment (garyhtou) [reply app/views/users/edit.html.erb]: I wonder if some users may mistake the placeholder as their pronouns being set to "they/them/theirs".

I think I like the placeholder that Manu has at the moment

**Pull request closed without merging**